### ISSUE TITLE
Move s3 path to common place

### ISSUE THREAD

**Issue by IRCody** [intelsdi-x/snap-pluginsync#15]
I noticed on one of the recent PRs submitted against the [disk-collector](https://github.com/intelsdi-x/snap-plugin-collector-disk/pull/19) the path to the binary in s3 is changed.

This will necessitate changing the path to the binary inside of the existing large tests since they hardcode this binary path. I wonder if we could have a function inside common.sh to build the path to the s3 binary, being provided certain information. This would allow future changes to the path to be done transparently to the plugin-author/large tests. 

I also wonder if there are other things that could be centralized in a similar way so that the pluginsync updates rarely result in additional updates needed to the plugin.

**Comment by kindermoumoute** [comment]
> I also wonder if there are other things that could be centralized in a similar way so that the pluginsync updates rarely result in additional updates needed to the plugin.

I think there is something to synchronize with the `examples` folder and with large tests. There are really few differences in the files. Here is a proposal.

## Examples

* `examples/tasks/`
Contains all the task examples, named `<name_task>.json` or `<name_task>.yml`.

* `examples/configs/`
Contains all the config file examples, same name than the task associated with `<name_task>.json` or `<name_task>.yml`. (this file is optional)

* `examples/docker-compose.yml`
The first lines of this file are can be sync. e.g.
```
version: '2'
services:
  main:
    container_name: runner
    image: intelsdi/snap:alpine
    environment:
      - SNAP_VERSION=latest
    volumes:
      - ${PLUGIN_SRC}:/${PLUGIN_DIRECTORY_NAME}
```

* `examples/run-examples.sh`
Source `scripts/common.sh`, verify docker is installed, docker-compose, run the snap container calling `scripts/examples.sh`and starting bash.

* `scripts/examples.sh`
This is the big part of the cake, this script will:
	* Pick the first task manifest (and config file if needed) from `examples/tasks`. OPTIONAL: make we can choose it.
	* "read" the task and download the last version of the plugins needed for running this tasks.
	* (Send the config to snapd.)
	* Load the plugins.
	* Start the selected task

* `scripts/common.sh`
Most of the function described above are called from `scripts/common.sh`.


## Large tests
Basically keeping the current files.

* `scripts/large_compose.sh`
Change docker_folder to `"${__proj_dir}/examples`.

* `scripts/large_tests.sh`
Source `scripts/examples.sh` and run the tests.

* `scripts/large_k8s.sh`
Change "1/1" dynamically to the number of container that should be loaded from `scripts/config/<plugin_name>-deployment.yml`.

### notes
* `scripts/config/<plugin_name>-deployment.yml` can't be "easily" sync.
* Maybe this is something to do with the skeleton generator.


All the above files could be synchronized.

**Comment by jcooklin** [comment]
@kindermoumoute I think your proposal looks good however I think it should be moved to a separate issue as a potential enhancement so we can have a conversation on it there.  I see what @IRCody is suggesting here as a separate, but related, enhancement that can/should be done separately from your larger proposal.

**Comment by mjbrender** [comment]
Agreed with @jcooklin here. @kindermoumoute, please make sure the awesome write up above will be considered in another issue, possibly in #16 if that makes sense to you. 

Hey @nanliu, I believe this was the due to the unavailable change to paths in our cloud hosting of binaries. If that's right, we now have a process that will prevent it from happening again @IRCody. 

Is there any other place we should mention the path changes Cody to make sure everyone is aware?

**Comment by kindermoumoute** [comment]
@mjbrender #16 was a general proposal of the above writing.

**Comment by mjbrender** [comment]
Upon syncing up with @IRCody, I see the need for wrapping the path to the binary in large tests. We may change the storage location in the long run.

**Comment by nanliu** [comment]
This has been resolved by #47

```ruby
    case name
    when 'mock' # TODO: revisit how we handle mock plugins
      url = "https://s3-us-west-2.amazonaws.com/snap.ci.snap-telemetry.io/snap/#{version}/linux/x86_64/#{plugin_name}2"
    when 'mock1', 'mock2', 'mock2-grpc', 'passthru', 'passthru-grpc', 'mock-file', 'mock-file-grpc'
      url = "https://s3-us-west-2.amazonaws.com/snap.ci.snap-telemetry.io/snap/#{version}/linux/x86_64/#{plugin_name}"
    else
      url = "https://s3-us-west-2.amazonaws.com/snap.ci.snap-telemetry.io/plugins/#{plugin_name}/#{version}/linux/x86_64/#{plugin_name}"
    end
```
